joern-cli/frontends/x2cpg/src/main/scala/io/joern/x2cpg/astgen/AstGenRunner.scala

```diff
@@ -181,9 +181,13 @@ trait AstGenRunnerBase(config: X2CpgConfig[?] & AstGenConfig[?]) {
         val parsed  = filterFiles(srcFiles, out)
         val skipped = skippedFiles(in, result.toList)
         DefaultAstGenRunnerResult(parsed, skipped)
+      case Failure(f) if !isRetry =>
+        logger.error(s"\t- running ${metaData.name} failed! Retrying...", f)
+        execute(out, true)
```

Comment:
This could run endlessly. We should have at least some max. number of runs value as param.